### ISSUE TITLE
[BUG] sole f85 showing as found but doesn't display/interact

### ISSUE THREAD

**Issue by cornelious2** [cagnulein/qdomyos-zwift#2968]
**Describe the bug**
QZ app doesn't display any data in the tiles 

even heart rate not displaying when using the Garmin companion app. Treadmill on 2.5

**To Reproduce**

**Expected behavior**
Expecting tiles to show speed incline etc and give ability to control sole f85 treadmill 

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: pixel 8 pro 
 - OS: android 15
 - Version: AP4A.241205.013.C1

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Debug log slightly too big even after restarting phone can email a Google drive link.
  
  
**Additional context**

**Comment by cornelious2** [comment]
Sent a drive link with the debug logs controlling the treadmill with the sole+ app to Roberto's email

**Comment by cagnulein** [comment]
[btsnoop_hci.log.zip](https://github.com/user-attachments/files/18280014/btsnoop_hci.log.zip)
added only the bluetooth log
i will check it asap

**Comment by cagnulein** [comment]
hi, disable the inclination experimental in qz under the sole settings in the treadmill options
let me know

**Comment by cornelious2** [comment]
It was able to connect, no heartrate, and the incline sometimes worked
sometimes didn't. I'll try to get a log on what is happening after work
with more details.

Thank you!

**Comment by cagnulein** [comment]
perfect! for the inclination, the min step inclination is 0.5% or 1%?

**Comment by cornelious2** [comment]
0-12 are the settings on my sole f85, it's one from 2021/2022 timeframe

**Comment by cagnulein** [comment]
yes but does accepts also inclination with the .5 steps? if not the issue is just the min step inclination to the treadmill options that must be to 1.

let me know

**Comment by cornelious2** [comment]
Whole numbers only from zero degree incline up to 15

**Comment by cagnulein** [comment]
perfect so follow my previous comment

**Comment by cornelious2** [comment]
Elevation and heart rate seems to be working(both from the treadmill and from the Garmin companion app). Speed seems to be read only, is that expected for this model?

**Comment by cagnulein** [comment]
It depends of the firmware of the treadmill. Are you able to change it from
the sole app directly?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 3 gen 2025 alle 05:05 cornelious2 ***@***.***>
ha scritto:

**Comment by cornelious2** [comment]
I used to be able to in 2022 but not anymore.

**Comment by cagnulein** [comment]
Did you update the firmware since then?

**Comment by cornelious2** [comment]
Yes unfortunately. Do you know which versions are supported? I can try to
push sole for a downgrade option. Currently on 2.5 I think 2.3 or 2.4 was
the version I was on back when I emailed you early 2022.  Either way sounds
like qz is working as intended att this point.

**Comment by cagnulein** [comment]
No unfortunately I don't have any knowledge about specific sole firmwares. Try to contact the manufacturer, I always suggest to never upgrade the firmware if everything is working :/

**Comment by cornelious2** [comment]
QZ working as intended with the described settings changed. Any other missing features is due to treadmill firmware update.
